flink-core/src/main/java/org/apache/flink/core/plugin/PluginManager.java

```diff
@@ -45,6 +46,13 @@
 	/** List of patterns for classes that should always be resolved from the parent ClassLoader. */
 	private final String[] alwaysParentFirstPatterns;
 
+	@VisibleForTesting
+	PluginManager() {
+		parentClassLoader = null;
+		pluginDescriptors = null;
+		alwaysParentFirstPatterns = null;
+	}
```

Comment:
I think the clean solution for providing a `TestingPluginManager` implementation would be to extract an interface from the `PluginManager` and then having multiple implementations for it.